languages/haskell/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,53 @@
-# How to implement an Haskell concept exercise
+# How to implement a Haskell concept exercise
 
-TODO: describe how to implement a concept exercise for the Haskell track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise for the Haskell track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Concept exercises should be modelled after the CIS 194 Haskell course at Penn University, either the [Spring 2013][spring-2013] or [Fall 2016][fall-2016] version.
+
+To implement a concept exercise,  the following files need to be created:
+
+<pre>
+languages
+└── Haskell
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .meta
+                |   ├── config.json
```

Comment:
I'm not sure we've yet documented that. Here is the relevant part from the research experiment documentation: https://github.com/exercism/research_experiment_1#test-messages

Reply:
OK, so the current example from the research experiment for exercise-specific configuration is related to specialising the output of a test-runner, which we currently haven't got configured. 

Would you recommend that we omit `config.json` until we have something to put in them at a point in time where a test-runner is made, or that we create stub files containing an empty JSON object, `{}`?

Reply:
I think I would recommend creating a stub file. The new `config.json` also has the option to specify the solution files and test file: https://github.com/exercism/research_experiment_1/blob/master/exercises/csharp-1-a/.meta/config.json. This is used in the experiment, and it might be beneficial to v3 too. @iHiD What do you think?

Reply:
I wasn't aiming at turning this into a principal discussion, but it seems that my questions keep unravelling things that are still not standardised within Exercism, and my personality seems to force itself to address those.

1. I suspect the term "solution files" is ambiguous:

   The `config.json` at the track-level also has `"solution_pattern": "example.*[.]hs"`, but this refers to the *example solution*, where I suspect your use of the word refers to the Visual Studio nomenclature where "solution" is synonymous with "project"? In [your `csharp-1-a` example](https://github.com/exercism/research_experiment_1/blob/master/exercises/csharp-1-a/.meta/config.json), doesn't the Csharp1A.cs of `"solution_files": ["Csharp1A.cs"]` contain a *stub*?

2. With regards to the Haskell track's usage of exercise-specific configuration related to automated testing, I think it would be easier to wait until someone (myself or @aimorris) begin to set up a test-runner, at which point in time we make sense of it.

I am content with either creating a stub `{}` file or omitting it, but am leaning towards omitting it when it serves no purpose yet. Just as `after.md` is tagged with "(optional)", it would make sense to do so for this one until a point in time where something goes into it.

Reply:
Sure, let's omit the `config.json` for now until there is a test runner. We can address the non-standardized issues later.

Reply:
Sounds good. Do you want me to omit it completely, or just add (optional)?

Reply:
How about putting something like "(omit for now)"? That will both indicate that it has a role, but also that we don't make use of it yet. And we won't have to explain under what conditions it is optional, which will simplify the document a bit.